src/state/chainstate.rs

```diff
@@ -66,16 +66,6 @@ unsafe impl Send for ChainstateManager {}
 unsafe impl Sync for ChainstateManager {}
 
 impl ChainstateManager {
-    pub fn new(chainman_opts: ChainstateManagerOptions) -> Result<Self, KernelError> {
```

Comment:
```rust 
impl ChainstateManager {
    pub fn builder(
        context: &Context, 
        data_dir: &str, 
        blocks_dir: &str
    ) -> Result<ChainstateManagerBuilder, KernelError> {
        ChainstateManagerBuilder::new(context, data_dir, blocks_dir)
    }
}
```
and we can do:
```rust 
ChainstateManager::builder(&context, &data_dir, &blocks_dir)?
    .worker_threads(4)
    .build()?
```

What do you think?

Reply:
Yeah, might make sense to implement a default builder() and new() method for convenience.